numpy/core/_add_newdocs.py

```diff
@@ -7970,62 +7970,169 @@ def luf(lamdaexpr, *args, **kwargs):
 #
 ##############################################################################
 
-add_newdoc('numpy.core.numerictypes', 'bool_',
-    """NumPy's Boolean type.  Character code: ``?``.  Alias: bool8""")
+def get_type(place, obj):
+    try:
+        return getattr(__import__(place, globals(), {}, [obj]), obj)
+    except AttributeError:
+        return None
+
+def type_aliases(place, aliases):
+    def type_aliases_gen():
+        for (alias, doc) in aliases:
+            alias_type = get_type(place, alias)
+            if alias_type is not None:
+                yield (alias_type, alias, doc)
+    return list(type_aliases_gen())
+
+integer_aliases = type_aliases('numpy.core.numerictypes', [
+    ('int8', '8-bit signed integer (-128 to 127)'),
+    ('int16', '16-bit signed integer (-32768 to 32767)'),
+    ('int32', '32-bit signed integer (-2147483648 to 2147483647)'),
+    ('int64', '64-bit signed integer (-9223372036854775808 to 9223372036854775807)'),
+    ('intp', 'Signed integer large enough to fit pointer, compatible with C ``intptr_t``'),
+    ])
+
+unsigned_integer_aliases = type_aliases('numpy.core.numerictypes', [
+    ('uint8', '8-bit unsigned integer (0 to 255)'),
+    ('uint16', '16-bit unsigned integer (0 to 65535)'),
+    ('uint32', '32-bit unsigned integer (0 to 4294967295)'),
+    ('uint64', '64-bit unsigned integer (0 to 18446744073709551615)'),
+    ('uintp', 'Unsigned integer large enough to fit pointer, compatible with C ``uintptr_t``'),
+    ])
+
+float_aliases = type_aliases('numpy.core.numerictypes', [
+    ('float16', '16-bit-precision floating-point number type: sign bit, 5 bits exponent, 10 bits mantissa'),
+    ('float32', '32-bit-precision floating-point number type: sign bit, 8 bits exponent, 23 bits mantissa'),
+    ('float64', '64-bit precision floating-point number type: sign bit, 11 bits exponent, 52 bits mantissa'),
+    ('float96', '96-bit extended-precision floating-point number type'),
+    ('float128', '128-bit extended-precision floating-point number type'),
+    ])
+
+complex_aliases = type_aliases('numpy.core.numerictypes', [
+    ('complex64', 'Complex number type composed of 2 32-bit-precision floating-point numbers'),
+    ('complex128', 'Complex number type composed of 2 64-bit-precision floating-point numbers'),
+    ('complex192', 'Complex number type composed of 2 96-bit extended-precision floating-point numbers'),
+    ('complex256', 'Complex number type composed of 2 128-bit extended-precision floating-point numbers'),
+    ])
+
+
+def add_newdoc_for_numeric_type(obj, character_code, fixed_aliases, possible_aliases, doc):
```

Comment:
Instead of passing the character code, you can just use `np.dtype(o).char` below

Reply:
Same here, actually. To do that, I have to import numpy while numpy is being imported? :-/

Reply:
Numpy is already full of import inter-dependencies. I don't have a simple description of the rules, but as long as there are no cycles at import time, things are usually ok